fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -331,7 +348,7 @@ public static Map<String, List<List<String>>> getPartitionList(Table table, Colu
             throws UserException {
         if (table.isNativeTableOrMaterializedView()) {
             return ((OlapTable) table).getListPartitionMap();
-        } else if (table.isHiveTable() || table.isHudiTable() || table.isIcebergTable()) {
+        } else if (table.isHiveTable() || table.isHudiTable() || table.isIcebergTable() || table.isPaimonTable()) {
```

Comment:
There are too many conditions to check, maybe we can refactor this to be better extensible later.

Reply:
Next pr will work on it